gateway/src/test/java/io/zeebe/gateway/api/topology/TopologyTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.gateway.api.topology;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.zeebe.gateway.api.util.GatewayTest;
+import io.zeebe.gateway.impl.broker.cluster.BrokerClusterStateImpl;
+import io.zeebe.gateway.protocol.GatewayOuterClass.BrokerInfo;
+import io.zeebe.gateway.protocol.GatewayOuterClass.Partition.PartitionBrokerHealth;
+import io.zeebe.gateway.protocol.GatewayOuterClass.TopologyRequest;
+import io.zeebe.gateway.protocol.GatewayOuterClass.TopologyResponse;
+import org.junit.Test;
+
+public class TopologyTest extends GatewayTest {
+
+  @Test
+  public void shouldResponseWithInitialUnhealthyPartitions() {
+    // when
+    final TopologyResponse topologyResponse = client.topology(TopologyRequest.newBuilder().build());
+
+    // then
+    assertThat(topologyResponse.getBrokersList())
+        .isNotEmpty()
+        .allSatisfy(
+            brokerInfo ->
+                assertThat(brokerInfo.getPartitionsList())
+                    .isNotEmpty()
+                    .allSatisfy(
+                        partitionPerBroker ->
+                            assertThat(partitionPerBroker.getHealth())
+                                .isEqualTo(PartitionBrokerHealth.UNHEALTHY)));
+  }
+
+  @Test
+  public void shouldUpdatePartitionHealthHealthy() {
+    // given
+    final BrokerClusterStateImpl topology =
+        (BrokerClusterStateImpl) brokerClient.getTopologyManager().getTopology();
+    topology.setPartitionHealthy(0, 1);
+
+    // when
+    final TopologyResponse topologyResponse = client.topology(TopologyRequest.newBuilder().build());
+
+    // then
+    final PartitionBrokerHealth health =
+        topologyResponse.getBrokers(0).getPartitions(0).getHealth();
+    assertThat(health).isEqualTo(PartitionBrokerHealth.HEALTHY);
+  }
+
+  @Test
+  public void shouldUpdatePartitionHealthUnhealthy() {
```

Comment:
I think this test is not necessary. You've changed a property of the topology and then again get the same object and alter it again. Just once is fine. Or am I missing something?

Reply:
There I'm thinking to test simultaneously changing the health of two partitions. All can work fine if the broker changes the health of one partition but not for many :) I'm don't sure that this needs to be tested but anyway, what do you think? :)

Reply:
That makes sense, but you can do that in one go. No need for the additional setup. (I understand marking 1 partition first as healthy may matter for your implementation, but that should not be part of the test IMO)

Reply:
Also make sure to change the name of the test method accordingly. Something like: `shouldUpdateMultiplePartitionHealths`

Reply:
Done :)